data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/GreaterThanOperator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class GreaterThanOperator implements Operator<Boolean> {
+    @Override
+    public Integer getSymbol() {
+        return DataPrepperExpressionParser.GT;
+    }
+
+    @Override
+    public Boolean eval(Object... args) {
+        checkArgument(args.length == 2, "Operands length needs to be 2.");
+        if ((args[0] instanceof Integer) && (args[1] instanceof Float)) {
+            return ((Integer) args[0]) > ((Float) args[1]);
+        } else if ((args[0] instanceof Float) && (args[1] instanceof Integer)) {
+            return ((Float) args[0]) > ((Integer) args[1]);
+        } else if ((args[0] instanceof Integer) && (args[1] instanceof Integer)) {
+            return ((Integer) args[0]) > ((Integer) args[1]);
+        } else if ((args[0] instanceof Float) && (args[1] instanceof Float)) {
+            return ((Float) args[0]) > ((Float) args[1]);
+        }
```

Comment:
Consider converting argument into [BigDecimal](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html) and simplifying the code for maintainability. The BigDecimal class gives complete control over rounding behavior and is an exact way of representing numbers.

Below applies to all number comparison operation.

```
 public Boolean eval(final Object... args) {
        checkArgument(args.length == 2, "Operands length needs to be 2.");
        return getBigDecimal(val1).compareTo(getBigDecimal(val1)) == 0;
    }

// this can be moved to some common class
public static BigDecimal getBigDecimal( final Object value ) {
        BigDecimal ret = null;
        if( value != null ) {
            if( value instanceof BigDecimal ) {
                ret = (BigDecimal) value;
            } else if( value instanceof Integer ) {
                ret = new BigDecimal( (Integer) value );
            } else if( value instanceof Float ) {
                ret = new BigDecimal( (Float) value );
            } else {
                throw new IllegalArgumentException/ClassCastException("Not possible to convert ["+value+"] from class "+value.getClass()+" into a BigDecimal.");
            }
        }
        return ret;
    }
```

Reply:
@dinujoh Thanks for the suggestion! Although it refactors similar lines of code into single method for better readability, I am a little worried about the overhead of converting number into BigDecimal (getBigDecimal): https://stackoverflow.com/questions/1378044/how-using-bigdecimal-would-affect-application-performance. Also on precision aspect, can we reserve this change until higher precision is indeed desired in use cases? @sbayer55 thoughts are welcome

Reply:
@sbayer55 can comment if precision is required or we can reserve it for later. If we want to support double then we would require precision.

Consider the following code which is much more readable and avoids if-else.
```
private static final Map<String, BiFunction<Object, Object, Boolean>> comparisonFunction = new HashMap<>() {
  {
       put(Integer.class.getName() + "_" + Integer.class.getName(), (a,b) -> (Integer)a > (Integer)b);
       put(Integer.class.getName() + "_" + Float.class.getName(), (a,b) -> (Integer)a > (Float)b);
       put(Float.class.getName() + "_" + Integer.class.getName(), (a,b) -> (Float)a > (Integer)b);
       put(Float.class.getName() + "_" + Float.class.getName(), (a,b) -> (Float)a > (Float)b);
  }
};

public boolean compare(final Object val1, final Object val2) {
    final BiFunction<Object, Object, Boolean> operation = comparisonFunction.get(constructKey(val1, val2));
   if(operation == null) {
      throw new IllegalArgumentException(val1 + " or " + val2 + " should be either Float or Integer");
   }
   return operation.apply(val1, val2);
}

// can be moved to common class
private String constructKey(final Object val1, final Object val2) {
   return val1.getClass().getName() + "_" + val2.getClass().getName();
}
```

Reply:
@dinujoh Thanks. This indeed looks more compact! I will make the change to see what others think.

Reply:
@dinujoh adding a tolerance to precision operators is out of scope for 1.3. I created issue (#1109)[https://github.com/opensearch-project/data-prepper/issues/1109] to track enhancing floating point operation support.

Reply:
@dinujoh I've heard BigDecimal is significantly slower compared to int/float operations. Do you know if there is any truth to that?